llvm/test/CodeGen/DirectX/length.ll

```diff
@@ -0,0 +1,107 @@
+; RUN: opt -S  -dxil-intrinsic-expansion  < %s | FileCheck %s --check-prefixes=CHECK,EXPCHECK
+; RUN: opt -S  -dxil-op-lower  < %s | FileCheck %s --check-prefixes=CHECK,DOPCHECK
+
+; ModuleID = 'D:\llvm-project\clang\test\CodeGenHLSL\builtins\length.hlsl'
+source_filename = "D:\\llvm-project\\clang\\test\\CodeGenHLSL\\builtins\\length.hlsl"
+target datalayout = "e-m:e-p:32:32-i1:32-i8:8-i16:16-i32:32-i64:64-f16:16-f32:32-f64:64-n8:16:32:64"
+target triple = "dxilv1.3-pc-shadermodel6.3-library"
+
+; Function Attrs: convergent noinline nounwind optnone
+define noundef float @"?test_length_half@@YA$halff@$halff@@Z"(float noundef %p0) #0 {
```

Comment:
test seems wrong half is in the name but input and return value are float.

Reply:
I generated this file by compiling length.hlsl and taking the dxil output of the first 4 functions, which were testing half, but with NO_HALF enabled. So, I just need to change the function names, the test doesn't seem wrong to me.

Reply:
You already have float tests, so you want half test cases for full coverage. the ret, input, and intrinsic invocation needs to be updated not the function name.